test/util.spec.js

```diff
@@ -688,16 +688,28 @@
         e: 5
       });
     });
-    return it('should return the merged object', function() {
+    it('should return the merged object', function() {
       var obj;
       obj = {
         a: 1,
         b: 2
       };
       return expect(AWS.util.update(obj, {
         c: 3
-      })).to.equal(obj);
+      })).to.eql({a: 1, b: 2, c: 3});
     });
+    it('should merge an object into undefined object', function() {
```

Comment:
Why should this work? I don't think updating an object that is `undefined` or `null` is expected, since this method is currently meant to mutate an existing object. Was this change needed to fix something?

Reply:
[here](https://github.com/aws/aws-sdk-js/blob/214b42460882384ab1ac5edd6ec48a6076f31f8f/lib/credentials/shared_ini_file_credentials.js#L181) is where is issue is and I changed to [`AWS.util.merge`](https://github.com/aws/aws-sdk-js/blob/214b42460882384ab1ac5edd6ec48a6076f31f8f/lib/util.js#L508). Actually this fix doesn't make sense, instead we should allow merge function to merge undefined or null